File: src/main/java/org/alham/alhamfirst/dto/user/UserDTO.java
package org.alham.alhamfirst.dto.user;

import lombok.*;
import org.alham.alhamfirst.common.enums.UserType;
import org.alham.alhamfirst.util.AESUtil;

@Getter
@Setter
@ToString
@NoArgsConstructor
@Builder
@AllArgsConstructor
public class UserDTO {


    private String id;
    private String name;
    private int age;
    private UserType userType;
    private String email;


    public static UserDTO getEmptyUser(){
        return new UserDTO();
    }

    public long getIdDecrypt() {
        return Long.parseLong(AESUtil.decrypt(this.id));
    }

}

File: src/main/java/org/alham/alhamfirst/dto/user/UserDTO.kt
package org.alham.alhamfirst.dto.user;

import lombok.*;
import org.alham.alhamfirst.common.enums.UserType;
import org.alham.alhamfirst.util.AESUtil;

import java.util.HashMap;
import java.util.Map;

//@Getter
//@Setter
//@ToString
//@NoArgsConstructor
//@Builder
//@AllArgsConstructor
data class UserDTO(
        var id : String = "",
        var name : String ="",
        var age : Int = 0,
        var userType : UserType = UserType.BASIC,
        var email : String = "",
        var statData : MutableMap<String,Int> = mutableMapOf()
) {


    fun getEmptyUser() :  UserDTO{
        return UserDTO();
    }

    fun getIdDecrypt() : Long{
        return AESUtil.decrypt(this.id).toLong();
    }

}
